crates/cargo-test-support/src/git.rs

```diff
@@ -175,6 +175,20 @@ where
     (git_project, repo)
 }
 
+/// Create a new git repository with a project.
+/// Returns both the Project and the git Repository but does not commit.
+pub fn new_repo_without_add_and_commit<F>(name: &str, callback: F) -> (Project, git2::Repository)
```

Comment:
imo the one test needing this should just do itself rather than creating this function. The heavy lifting of these functions is setting up that commit which this doesn't need